File: src/main/com/dune/ide/files/DuneFile.java
package com.dune.ide.files;

import com.dune.DuneLanguage;
import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

public class DuneFile extends PsiFileBase {
    public DuneFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, DuneLanguage.INSTANCE);
    }

    @NotNull @Override public FileType getFileType() {
        return DuneFileType.INSTANCE;
    }

    @NotNull @Override public String toString() {
        return DuneFileType.INSTANCE.getDescription();
    }
}

File: src/main/kotlin/com/dune/language/psi/files/DuneFile.kt
package com.dune.language.psi.files

import com.dune.ide.files.DuneFileType
import com.dune.language.DuneLanguage
import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

class DuneFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, DuneLanguage) {
    override fun getFileType(): FileType = DuneFileType
    override fun toString(): String = "Dune File"
}